libcni/api.go

```diff
@@ -172,8 +177,13 @@ func (c *CNIConfig) addOrGetNetwork(command, name, cniVersion string, net *Netwo
 	if err != nil {
 		return nil, err
 	}
-
-	return invoke.ExecPluginWithResult(pluginPath, newConf.Bytes, c.args(command, rt), c.exec)
+	if rt.Timeout.Nanoseconds() > 0 {
+		ctx, cancel = context.WithTimeout(context.Background(), rt.Timeout)
+		defer cancel()
+	} else {
+		ctx = context.Background()
+	}
```

Comment:
If you do the above suggestion and plumb the Context down through to addOrGetNetwork, then you don't need to do any of this since the caller has already done it for you.